modules/apps/callbacks/ibc_middleware.go

```diff
@@ -0,0 +1,233 @@
+package ibccallbacks
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	capabilitytypes "github.com/cosmos/ibc-go/modules/capability/types"
+	"github.com/cosmos/ibc-go/v7/modules/apps/callbacks/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v7/modules/core/05-port/types"
+	ibcexported "github.com/cosmos/ibc-go/v7/modules/core/exported"
+)
+
+var (
+	_ porttypes.Middleware            = (*IBCMiddleware)(nil)
+	_ porttypes.PacketDataUnmarshaler = (*IBCMiddleware)(nil)
+)
+
+// IBCMiddleware implements the ICS26 callbacks for the ibc-callbacks middleware given
+// the underlying application.
+type IBCMiddleware struct {
+	app     types.PacketUnmarshalerIBCModule
+	channel porttypes.ICS4Wrapper
+
+	contractKeeper types.ContractKeeper
+}
+
+// NewIBCMiddleware creates a new IBCMiddlware given the keeper and underlying application
+func NewIBCMiddleware(
+	app porttypes.IBCModule,
+	channel porttypes.ICS4Wrapper,
```

Comment:
channel? I would recommend updating the name as this can be another application (such as ics29)

Reply:
done